File: src/main/java/no/nav/familie/ks/sak/config/ApplicationConfig.java
package no.nav.familie.ks.sak.config;

import no.nav.familie.log.filter.LogFilter;
import no.nav.security.token.support.client.spring.oauth2.EnableOAuth2Client;
import no.nav.security.token.support.spring.api.EnableJwtTokenValidation;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.boot.SpringBootConfiguration;
import org.springframework.boot.autoconfigure.domain.EntityScan;
import org.springframework.boot.web.embedded.jetty.JettyServletWebServerFactory;
import org.springframework.boot.web.servlet.FilterRegistrationBean;
import org.springframework.boot.web.servlet.server.ServletWebServerFactory;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.data.jpa.repository.config.EnableJpaAuditing;
import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
import org.springframework.retry.annotation.EnableRetry;

@SpringBootConfiguration
@EnableJpaAuditing
@EnableJpaRepositories({"no.nav.familie"})
@EntityScan({"no.nav.familie"})
@ComponentScan({"no.nav.familie"})
@EnableRetry
@EnableJwtTokenValidation
@EnableOAuth2Client(cacheEnabled = true)
public class ApplicationConfig {

    private static final Logger log = LoggerFactory.getLogger(ApplicationConfig.class);

    @Bean
    ServletWebServerFactory servletWebServerFactory() {

        JettyServletWebServerFactory serverFactory = new JettyServletWebServerFactory();

        serverFactory.setPort(8083);

        return serverFactory;
    }

    @Bean
    public FilterRegistrationBean<LogFilter> logFilter() {
        log.info("Registering LogFilter filter");
        final FilterRegistrationBean<LogFilter> filterRegistration = new FilterRegistrationBean<>();
        filterRegistration.setFilter(new LogFilter());
        filterRegistration.setOrder(1);
        return filterRegistration;
    }
}

File: src/main/kotlin/no/nav/familie/ks/sak/config/ApplicationConfig.kt
package no.nav.familie.ks.sak.config

import no.nav.familie.log.NavSystemtype
import no.nav.familie.log.filter.LogFilter
import no.nav.familie.log.interceptor.ConsumerIdClientInterceptor
import no.nav.familie.sikkerhet.context.FamilieFellesSpringSecurityKonfigurasjon
import org.slf4j.LoggerFactory
import org.springframework.boot.SpringBootConfiguration
import org.springframework.boot.context.properties.ConfigurationPropertiesScan
import org.springframework.boot.persistence.autoconfigure.EntityScan
import org.springframework.boot.web.servlet.FilterRegistrationBean
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.Import
import org.springframework.retry.annotation.EnableRetry
import org.springframework.scheduling.annotation.EnableScheduling

@SpringBootConfiguration
@EntityScan("no.nav.familie.prosessering", ApplicationConfig.PAKKENAVN)
@ComponentScan(
    "no.nav.familie.prosessering",
    "no.nav.familie.unleash",
    "no.nav.familie.felles.tokenklient",
    ApplicationConfig.PAKKENAVN,
)
@EnableRetry
@ConfigurationPropertiesScan
@EnableScheduling
@Import(
    ConsumerIdClientInterceptor::class,
    FamilieFellesSpringSecurityKonfigurasjon::class,
)
class ApplicationConfig {
    @Bean
    fun logFilter(): FilterRegistrationBean<LogFilter> {
        log.info("Registering LogFilter filter")

        return FilterRegistrationBean<LogFilter>().apply {
            setFilter(LogFilter(systemtype = NavSystemtype.NAV_SAKSBEHANDLINGSSYSTEM))
            order = 1
        }
    }

    companion object {
        private val log = LoggerFactory.getLogger(ApplicationConfig::class.java)
        const val PAKKENAVN = "no.nav.familie.ks.sak"
    }
}
